examples/interaction/tools/persistent_selections.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+
+from bokeh.models import BoxSelectTool, LassoSelectTool, PolySelectTool
+from bokeh.plotting import figure, show
+
+N = 2000
+x = np.random.random(size=N) * 100
+y = np.random.random(size=N) * 100
+radii = np.random.random(size=N) * 1.5
+colors = np.array([ [r, g, 150] for r, g in zip(50 + 2*x, 30 + 2*y) ], dtype="uint8")
+
+TOOLS="hover,crosshair,pan,wheel_zoom,zoom_in,zoom_out,box_zoom,undo,redo,reset,save,box_select,poly_select,lasso_select,help"
```

Comment:
For the most part they aren't unrelated. For example, crosshair and hover tools are important to show how UI events are propagated. Pan and zoom tools are useful for showing how overlays are persisted between different interactions. I refactored the example to get rid of the list, but otherwise most of those tools should stay, maybe with some commentary what to look for.

Reply:
The updated version seems better, thanks